htdocs/stripe/class/stripe.class.php

```diff
@@ -402,8 +402,8 @@ public function getPaymentIntent($amount, $currency_code, $tag, $description = '
     				if (! $paymentintentalreadyexists)
     				{
 	    				$now=dol_now();
-	    				$sql = "INSERT INTO " . MAIN_DB_PREFIX . "prelevement_facture_demande (date_demande, fk_user_demande, ext_payment_id, fk_facture, sourcetype, entity, ext_payment_site)";
-	    				$sql .= " VALUES ('".$this->db->idate($now)."', '0', '".$this->db->escape($paymentintent->id)."', ".$object->id.", '".$this->db->escape($object->element)."', " . $conf->entity . ", '" . $service . "')";
+	    				$sql = "INSERT INTO " . MAIN_DB_PREFIX . "prelevement_facture_demande (fk_soc, date_demande, fk_user_demande, ext_payment_id, fk_facture, sourcetype, entity, ext_payment_site)";
+	    				$sql .= " VALUES ('".$object->socid."', '".$this->db->idate($now)."', '0', '".$this->db->escape($paymentintent->id)."', ".$object->id.", '".$this->db->escape($object->element)."', " . $conf->entity . ", '" . $service . "')";
```

Comment:
Why adding this field ?
We already have the if of object so, we are already able to know the socid.
If a intent is done on invoice X, it is on customer of invoice X,we don't need to duplicate the information that creates non consistent data in database.

Reply:
Free payment have no object but we need to attach them to a thirdparty for use save card or other things

Reply:
By definition free payment are "anonymous payment" not linked to a thirdparty.
Here you fill the value with $object->socid. But because free payment are used to make payment without dedicated object, this field will also always be empty. In fact, $object will be empty when making free payment.

We can enhance free payment to support free payment on a particular thirdparty, but, for that, we must
- add field into table and into migration
- change the code to pass the thirdparty id as a parameter of free payment url and reuse this parameter to fill the field $fk_soc instead of$object->fk_soc that is not defined.